src/flow.ts

```diff
@@ -327,7 +327,8 @@ export class Flow {
       }
       local = parentFunction.addLocal(type);
     } else {
-      if (temps !== null && temps.length > 0) {
+      if (temps !== null && temps.length > 0
+          && temps[temps.length-1].type === type) {
```

Comment:
What are the types here when the conflict happens? Asking because the `temps` array should already only contain matching types (see the switch above), so this is a bit unexpected.

Reply:
it's because custom class types becomes usize which resolves to i32 in the switch above. and so the temps contains both i32 and other class types.

Reply:
Hmm, thinking about this a little more, what may be happening there is that a temporary local is not properly blocked to be not re-used during compilation. The mechanism there is to obtain a temporary local, use it, and free it again so it can be reused later on. Now, if the free happens too early, things may go wrong, which may be the root cause.

Reply:
Yeah I was suspecting I’m just workin around a more underlying problem here. Will see if freeing happens too early.

Reply:
I think I got it now @dcodeIO , `flow.freeTempLocal(tempTarget);` was called twice in compiler / makeAssignment